### PR TITLE
[4.0] RavenDB-10959 & RavenDB-10960

### PR DESCRIPTION
**Pull request by grisha-kotler**
http://issues.hibernatingrhinos.com/issue/RavenDB-10959
http://issues.hibernatingrhinos.com/issue/RavenDB-10960

### CODE REVIEW

File: src/Raven.Client/Documents/Operations/Backups/RestoreBackupConfiguration.cs [hunk 1]
```diff
@@ -11,5 +11,7 @@ public class RestoreBackupConfiguration
         public string DataDirectory { get; set; }
 
         public string EncryptionKey { get; set; }
+
+        public bool DisableOngoingTasks { get; set; } = true;
```
Comment (ayende) [inline src/Raven.Client/Documents/Operations/Backups/RestoreBackupConfiguration.cs]: This should default to false.

File: src/Raven.Server/Web/System/AdminDatabasesHandler.cs [hunk 2]
```diff
@@ -1005,41 +1005,51 @@ public Task CompactDatabase()
...
+                                }
+
+                                await compactDatabaseTask.Execute(onProgress, overallResult);
                                 overallResult.Processed = true;
-                                return overallResult;
-                            }
 
-                            await compactDatabaseTask.Execute(onProgress, overallResult);
-                            overallResult.Processed = true;
+                                overallResult.SizeAfterCompactionInMb = CalculateStorageSizeInBytes(compactSettings.DatabaseName).Result / 1024 / 1024;
```
Comment (ppekrol) [inline src/Raven.Server/Web/System/AdminDatabasesHandler.cs]: use await?

File: src/Raven.Server/Web/System/AdminDatabasesHandler.cs [hunk 3]
```diff
@@ -1005,41 +1005,51 @@ public Task CompactDatabase()
...
-                            var overallResult = new CompactionResult(compactSettings.DatabaseName);
-
-                            // first fill in data 
-                            foreach (var indexName in compactSettings.Indexes)
+                            using (token)
                             {
-                                var indexCompactionResult = new CompactionResult(indexName);
-                                overallResult.IndexesResults.Add(indexName, indexCompactionResult);
-                            }
+                                var before = CalculateStorageSizeInBytes(compactSettings.DatabaseName).Result / 1024 / 1024;
```
Comment (ppekrol) [inline src/Raven.Server/Web/System/AdminDatabasesHandler.cs]: use await?

Comment (grisha-kotler) [reply src/Raven.Server/Web/System/AdminDatabasesHandler.cs]: done

File: src/Raven.Server/Web/System/AdminDatabasesHandler.cs (hunk 2 above)
Comment (grisha-kotler) [reply src/Raven.Server/Web/System/AdminDatabasesHandler.cs]: done

File: src/Raven.Client/Documents/Operations/Backups/RestoreBackupConfiguration.cs (hunk 1 above)
Comment (grisha-kotler) [reply src/Raven.Client/Documents/Operations/Backups/RestoreBackupConfiguration.cs]: done

**Comment by grisha-kotler** [comment]
fixed

**Pull request merged**